### PR TITLE
ticket 145

### PR DESCRIPTION
**Pull request by merlouduron**
# Pull Request Review Comments

## Overview
- **List of finished task**:
    - [x] Added CRUD function
    - [x] Add overwrite function
    - [x] Validation for multiple save
    - [x]  Added a hover for selecting project name to identify which one to select base on client name
    - [x] Fix SCSS and classnames
    - [x] Added search function
    - [x] Disabled project_type, client and business_division inputs.

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- Add any additional comments or information for the reviewer here.

### CODE REVIEW

File: client/src/translations/financial_words_jp.json [hunk 1]
```diff
@@ -55,6 +55,10 @@
   "profitAndLossPlaninng": "PL 計画",
   "purchases": "仕入高",
   "purchase": "仕入高",
+  "projectSalesResultsId": "アイド",
+  "projectSalesResultsList": "案件売上実績一覧",
+  "projectSalesResultsRegistration": "案件売上実績登録",
+  "projectSalesResults" : "プロジェクト売上実績",
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply client/src/translations/financial_words_jp.json]: Fixed

File: client/src/translations/financial_words_jp.json [hunk 2]
```diff
@@ -55,6 +55,10 @@
   "profitAndLossPlaninng": "PL 計画",
   "purchases": "仕入高",
   "purchase": "仕入高",
+  "projectSalesResultsId": "アイド",
+  "projectSalesResultsList": "案件売上実績一覧",
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply client/src/translations/financial_words_jp.json:59]: Fixed

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+                  <div>
+                    <div className='columns is-mobile'>
+                      <div className='column'>
+                        {isEditing ? (
+                          <div className='editScroll'>
+                            <table className='table is-bordered is-hoverable'>
+                              <thead>
+                                <tr className='projectSalesResultsList_table_title '>
+                                  <th className='projectSalesResultsList_table_title_content_vertical has-text-centered'>
+                                    {translate('projectSalesResultsId', language)}
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx]: Fixed

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx [hunk 4]
```diff
@@ -0,0 +1,793 @@
...
+                            {translate('projectName', language)}
+                          </label>
+                          {enable ? (
+                            <select
+                              className='projectSalesResultsRegistration_select-option'
+                              name='project_name'
+                              value={form.project_name}
+                              onChange={(e) => handleChange(index, e)}
+                            >
+                              <option value=''>-- Reset --</option>
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx]: Fixed.

File: client/src/translations/financial_words_jp.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/translations/financial_words_jp.json:59]: Comma was left in - please remove! Edit & List Screen

案件売上実績編集,

Screenshot 2024-11-21 at 09 29 13

[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/translations/financial_words_jp.json:59]: Also I know its not this ticket but could you fix thiese please!

1.
expensesEdit

Screenshot 2024-11-21 at 09 56 09

2. Remove Space in between kanji.
Screenshot 2024-11-21 at 09 56 41

3. Add an 's' to result in "Expense Result" in English Mode
Screenshot 2024-11-21 at 09 57 08

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Please see new comments. Just some small changes.

File: client/src/translations/financial_words_jp.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/translations/financial_words_jp.json:59]: Fixed

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx [hunk 5]
```diff
@@ -0,0 +1,854 @@
...
+                      <div className='projectSalesResultsRegistration_left-form-div projectSalesResultsRegistration_calc'>
+                        <div className='projectSalesResultsRegistration_year-div'>
+                          <label className='projectSalesResultsRegistration_year'>{translate('year', language)}</label>
+                          <select
+                            className='projectSalesResultsRegistration_select-option'
+                            name='year'
+                            value={form.year}
+                            onChange={(e) => handleChange(index, e)}
+                          >
+                            <option value=''></option>
```
Comment (edward-stalley-work) [inline client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx:557]: May be better to add  "Select Year" in Japanese and English.

年選択

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx [hunk 6]
```diff
@@ -0,0 +1,854 @@
...
+                          <label className='projectSalesResultsRegistration_month'>
+                            {translate('month', language)}
+                          </label>
+                          <select
+                            className='projectSalesResultsRegistration_select-option'
+                            name='month'
+                            value={form.month}
+                            onChange={(e) => handleChange(index, e)}
+                          >
+                            <option value=''></option>
```
Comment (edward-stalley-work) [inline client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx:644]: May be better to add  "Select Month" in Japanese and English.

月選択

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx [hunk 7]
```diff
@@ -0,0 +1,854 @@
...
+                            {translate('projectName', language)}
+                          </label>
+                          {enable ? (
+                            <select
+                              className='projectSalesResultsRegistration_select-option'
+                              name='project_name'
+                              value={form.project_name}
+                              onChange={(e) => handleChange(index, e)}
+                            >
+                              <option value=''></option>
```
Comment (edward-stalley-work) [inline client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx:722]: May be better to add  "Select Project" in Japanese and English.

案件選択

Comment (edward-stalley-work) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx:722]: Also, when you select a project then go back and deselect the other inputs stay filled.
They need to be cleared.

Example:
Screenshot 2024-11-25 at 10 23 03

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx [hunk 8]
```diff
@@ -0,0 +1,645 @@
...
+    setIsCRUDOpen(false)
+  }
+
+  const handleConfirm = async () => {
+    if (!token) {
+      window.location.href = '/login'
+      return
+    }
+
+    deleteProjectSalesResults(deleteProjectsId, token)
```
Comment (edward-stalley-work) [inline client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx:257]: When delete is pressed the record remains on page until a manual refresh.

Record was deleted but still displayed.

Screenshot 2024-11-25 at 10 28 35

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx [hunk 9]
```diff
@@ -0,0 +1,645 @@
...
+        [name]: value,
+      }
+
+      setFormProjects(updatedProjectsData)
+      return updatedProjectsData
+    })
+  }
+
+  const handleSubmit = async () => {
+    const getModifiedFields = (original, updated) => {
```
Comment (edward-stalley-work) [inline client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx:121]: No validation check on empty inputs.
This is fine however as I will pull this into my ticket and add the updated validation for front end I have been working on.

-Ed will handle in another ticket

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Please see comments

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx (hunk 5 above)
Comment (merlouduron) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx:557]: Fixed

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx (hunk 6 above)
Comment (merlouduron) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsRegistration.tsx:644]: Fixed.

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx (hunk 8 above)
Comment (merlouduron) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx:257]: Fixed.

File: client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx (hunk 9 above)
Comment (merlouduron) [reply client/src/pages/ProjectSalesResults/ProjectSalesResultsListAndEdit.tsx:121]: Thank you Ed. 🙇

File: client/src/translations/financial_words_jp.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/translations/financial_words_jp.json:60]: The above has not been addressed.
Screenshot 2024-11-25 at 15 01 02

[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/translations/financial_words_jp.json:60]: Please fix the text 

Screenshot 2024-11-25 at 15 07 31

Screenshot 2024-11-25 at 15 07 19

[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/translations/financial_words_jp.json:60]: fixed

**Review by edward-stalley-work — APPROVED** [review]

**Pull request merged**